.github/workflows/ci.yml

```diff
@@ -0,0 +1,443 @@
+      - name: Move gems into separate directory before persisting
+        run: |
+          mkdir -p artifacts/gem
+          mv pkg/*.gem artifacts/gem
+      - uses: actions/upload-artifact@v4
+        with:
+          name: gem-${{ matrix.platform }}
+          path: artifacts/gem
+
+  install-windows-mingw:
+    needs:
+      - cross-compile
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby-version:
+          - 2.7
+          - '3.0'
+
+    name: Install on Windows (MingW)
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+
+      - name: Download precompiled gem
+        uses: actions/download-artifact@v4
+        with:
+          name: gem-x64-mingw32
+
+      - name: Install native gem and test if TinyTDS loads
+        shell: pwsh
+        run: |
+          $rubyArchitecture = (ruby -e "puts RbConfig::CONFIG['arch']").Trim()
+          $gemVersion = (Get-Content VERSION).Trim()
+          $gemToInstall = "./tiny_tds-$gemVersion-$rubyArchitecture.gem"
+          
+          Write-Host "Looking to install $gemToInstall"
+          gem install --local "$gemToInstall"
+          
+          ruby -e "require 'tiny_tds'; puts TinyTds::Gem.root_path"
+          exit $LASTEXITCODE
+
+  test-windows-mingw:
+    needs:
+      - cross-compile
+    strategy:
+      fail-fast: false
+      matrix:
+        mssql-version:
+          #- 2017
+          - 2019
+          - 2022
+        ruby-version:
+          - 2.7
+          - '3.0'
+
+    name: Test on Windows (MingW)
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
+
+      - name: Download precompiled gem
+        uses: actions/download-artifact@v4
+        with:
+          name: gem-x64-mingw32
+
+      - name: Install native gem and restore cross-compiled code from it
+        shell: pwsh
+        run: |
+          $rubyArchitecture = (ruby -e "puts RbConfig::CONFIG['arch']").Trim()
+          $gemVersion = (Get-Content VERSION).Trim()
+          $gemToInstall = "./tiny_tds-$gemVersion-$rubyArchitecture.gem"
+
+          Write-Host "Looking to install $gemToInstall"
+          gem install --local --install-dir=./tmp "$gemToInstall"
+
+          # Restore precompiled code
+          $source = (Resolve-Path ".\tmp\gems\tiny_tds-$gemVersion-$rubyArchitecture\lib\tiny_tds").Path
+          $destination = (Resolve-Path ".\lib\tiny_tds").Path
+          Get-ChildItem $source -Recurse -Exclude "*.rb" | Copy-Item -Destination {Join-Path $destination $_.FullName.Substring($source.length)}
+
+          # Restore ports
+          Copy-Item -Path ".\tmp\gems\tiny_tds-$gemVersion-$rubyArchitecture\ports" -Destination "." -Recurse
+
+      - name: Setup MSSQL
+        uses: potatoqualitee/mssqlsuite@v1.7
```

Comment:
Are we able to use service containers here instead?

Reply:
fwiw, there's also this https://github.com/ankane/setup-sqlserver/

Reply:
Linux service containers are not available on [Windows for GitHub Actions](https://github.com/orgs/community/discussions/25491). On Linux, the referenced action will use containers.

re: the action from ankane: would also work, also misses support for MSSQL 2017.